server/aesdsocket.c

```diff
@@ -0,0 +1,577 @@
+/*
+*   @file aesdsocket.c
+*
+*   @author Cedric Camacho
+*   @date Sep 18, 2024
+*
+*   @brief socket server code
+*/
+
+
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include <syslog.h>
+
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+
+#include <pthread.h>
+
+#include <time.h>
+
+#include "server.h"
+#include "queue.h"
+
+
+// #define USE_PRINT_DBUG
+
+#define IPV4            (0)
+#define IPV6            (1)
+
+#define SOCK_PORT       ("9000")
+#define IP_FORMAT       (IPV4)
+
+#if(IP_FORMAT)
+#define SOCK_DOMAIN     (PF_INET6)
+#define AI_FAM          (AF_INET6)
+
+#else
+#define SOCK_DOMAIN     (PF_INET)
+#define AI_FAM          (AF_INET)
+
+#endif
+
+#define BACKLOG_CNCTS   (5)
+#define DEFAULT_FILE    ("/var/tmp/aesdsocketdata")
+#define DELETE_FILE     (true)
+#define STRG_AVILBL     (64)
+
+#define TSTAMP_LENGTH   (100) // stamp format: day(5 max) month(3) year(4) time(HH:MM:SS)(8)\n = 21 (not including any spaces,etc)
+#define UPDATE_TIME     (10) // in seconds
+// #define US_TO_S         (1e6) // micro seconds
+// #define MS_TO_S         (1e3) // milliseconds
+
+#define SYSTEM_ERROR    (-1)
+
+
+TAILQ_HEAD(connection_head, connection_params_s) thread_list;
+
+
+pthread_mutex_t file_mutex;
+
+int accpt_fd, sock_fd, w_file_fd;
+static volatile sig_atomic_t sig_rec = false;
+
+pthread_t cleanup_thread_ID = 0;
+pthread_t timestamp_thread_ID = 0;
+
+/*
+*   @brief cleans up file descriptors and created file
+*
+*   @param[in] accept_fd
+*       - file descriptor of the accepted connection with client
+*
+*   @param[in] socket_fd
+*       - file descriptor of socket 
+*
+*   @param[in] file_fd
+*       - file descriptor of file
+*/
+static void cleanup(bool file_delete, int accept_fd, int socket_fd, int file_fd){
+
+    
+
+    if(accept_fd){
+        close(accept_fd);

```

Comment:
Getting this warning from valgrind here:
```
==28562== Warning: invalid file descriptor -1 in syscall close()
==28562==    at 0x487F45B: close (close.c:27)
==28562==    by 0x10B9C4: cleanup (aesdsocket.c:96)
==28562==    by 0x10B9C4: app_shutdown (aesdsocket.c:165)
==28562==    by 0x10A9F3: main (aesdsocket.c:573)
--28562-- Discarding syms at 0xde995e0-0xdeaa055 in /usr/lib/x86_64-linux-gnu/libgcc_s.so.1 (have_dinfo 1)
==28562==
```
I think you need to check for -1 on these close() calls.